src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -75,7 +76,77 @@ internal static int CompareStringIgnoreCaseNonAscii(ref char strA, int lengthA,
             return OrdinalCasing.CompareStringIgnoreCase(ref strA, lengthA, ref strB, lengthB);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Vector128EqualsIgnoreCaseAscii(Vector128<sbyte> vec1, Vector128<sbyte> vec2)
+        {
+            Debug.Assert(Vector128AllAscii(vec1.AsUInt16()));
+            Debug.Assert(Vector128AllAscii(vec2.AsUInt16()));
+            Vector128<sbyte> tmp1 = Vector128.Create((sbyte)0x3f) + vec1;
+            Vector128<sbyte> tmp2 = Vector128.Create((sbyte)0x3f) + vec2;
+            tmp1 = Vector128.LessThan(Vector128.Create((byte)0x99).AsSByte(), tmp1);
+            tmp2 = Vector128.LessThan(Vector128.Create((byte)0x99).AsSByte(), tmp2);
+            tmp1 = Vector128.AndNot(Vector128.Create((sbyte)0x20), tmp1);
+            tmp2 = Vector128.AndNot(Vector128.Create((sbyte)0x20), tmp2);
+            return ((tmp1 + vec1) ^ (tmp2 + vec2)) == Vector128<sbyte>.Zero;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Vector128AllAscii(Vector128<ushort> vec1)
+        {
+            return (vec1 & Vector128.Create(unchecked((ushort)~0x007F))) == Vector128<ushort>.Zero;
+        }
+
+        private static bool EqualsIgnoreCase_Vector128(ref char charA, ref char charB, int length)
+        {
+            Debug.Assert(length >= Vector128<ushort>.Count);
+            Debug.Assert(Vector128.IsHardwareAccelerated);
+
+            nuint lengthU = (nuint)length;
+            nuint lengthToExamine = lengthU - (nuint)Vector128<ushort>.Count;
+            nuint i = 0;
+            Vector128<ushort> vec1;
+            Vector128<ushort> vec2;
+            do
+            {
+                vec1 = Vector128.LoadUnsafe(ref Unsafe.As<char, ushort>(ref charA), i);
```

Comment:
Do the cast from `char -> ushort` once before the loop, then it's easier to read.
No perf-difference should occur.

Reply:
I think it usually makes sense when we have multiple `Unsafe.As` in the loop, here it only happens once so we'll only add two more lines of code here, will it increase code readability?

What I'd love us to do instead - add `char` overload to `Vector128.LoadUnsafe` (can be internal) - it should clean up a lot of similar cases.

Reply:
> add `char` overload to `Vector128.LoadUnsafe` (can be internal)

👍🏻 but please as `public`, as the `Unsafe.As` reinterpretation is tedious.